### DISCUSSION TITLE
How to contribute?

### DISCUSSION THREAD

**Discussion by zainulhassan815** [shreyashsaitwal/rush-cli#33] [Q&A]
Are there any rules or recommended way to contribute to this repository?

**Comment by shreyashsaitwal** [comment]
Umm, as of now, not really. But as long as your contribution...
* doesn't add too much of maintenance cost, and
* is a new feature that I'm already working on, but your implementation isn't better than mine,

...then I'm totally open to accepting it. If you already have some ideas to contribute, go ahead and do it! If not, then any help with current open issues and/or improving the documentation would be highly appreciated.

Also, if you have any questions regarding Rush's internal workings or anything related, feel free to ask them.

**Reply by zainulhassan815** [reply]
>     * is a new feature that I'm already working on, but your implementation isn't better than mine,

I guess you should create a pr with task list or features that are implementing. Then anyone could easily discuss and give their suggestions regarding that particular feature/task.

**Comment by zainulhassan815** [comment]
Got it. Thanks. So I do have better understanding of dart because of flutter but I never made console apps with it. But seems like dart is pretty handy in console apps too. I played a little bit with the code. I actually got basic understanding of how things are working. Plus it would be great if you could provide some helpful guide or some tutorial that you might have used while learning to develop console apps with dart. That would be helpful.

**Comment by shreyashsaitwal** [comment]
You can find many resources on building CLI apps with Dart on the internet. [This one](https://dart.dev/tutorials/server/cmdline) from the Dart docs is pretty good.

**Comment by zainulhassan815** [comment]
Can you explain a little bit about the dev branch? I mean currently the [PR#34](https://github.com/shreyashsaitwal/rush-cli/pull/34) was merged with the stable branch as it didn't have too many commits or features. But as for the new features or fixes, I don't think that the stable branch would be suitable. I guess it would be better to come up with a proper way to create new PRs in order to easily maintain code.

**Reply by shreyashsaitwal** [reply]
The dev branch is currently out of sync with the main branch, the changes made in the last couple of releases aren't merged in it. Along with that, it also has its own changes which consist of various new features and a lot of refactored code that was meant to land in Rush 2.0 development of which is currently on a halt for months. 

You're right, new features/fixes shouldn't be directly merged in the main branch. That's the reason why there was a dev branch in the first place. Anyways, I have renamed the old dev branch to 2.0, and have created a new dev branch from main. Please create any new PRs against it.

**Reply by zainulhassan815** [reply]
Awesome. Now it would be alot better.
